site/cypress/e2e/Cypress-Gradeable/late_submission_warning_messages.spec.js

```diff
@@ -0,0 +1,371 @@
+        cy.get('#create-gradeable-btn').click();
+
+        //Date page
+        cy.get('#page_5_nav').click();
+        cy.get('#date_ta_view')
+            .clear()
+            .type('1992-06-15')
+            .type('{enter}');
+        cy.get('#save_status', {timeout:20000}).should('have.text', 'All Changes Saved');
+
+        cy.get('#date_submit')
+            .clear()
+            .type('2004-12-18')
+            .type('{enter}');
+        cy.get('#save_status', {timeout:20000}).should('have.text', 'All Changes Saved');
+        cy.get('#late_days')
+            .clear()
+            .type(3)
+            .type('{enter}');
+        cy.get('#save_status', {timeout:20000}).should('have.text', 'All Changes Saved');
+        //Create team
+        cy.logout();
+        cy.login('student');
+        cy.visit(['sample', 'gradeable', team_gradeable, 'team']);
+        cy.get('#create_new_team').click();
+        cy.get('#invite_id').type('aphacker').type('{enter}');
+        cy.logout();
+        cy.login('aphacker');
+        cy.visit(['sample', 'gradeable', team_gradeable, 'team']);
+        cy.get('#accept_invitation').click();
+    });
+
+    it('should show a confirmation message for the first submission', () => {
+        //Part 1/2 of a test case
+        //The first submission will be done 2 days after the due date and use 2 valid late days for each team member
+        cy.login('instructor');
+        giveExtentions(team_gradeable);
+        giveLateDays(getCurrentTime('threeDaysAgo'),'student',3); //this is important for part 2/2
+        giveLateDays(getCurrentTime('threeDaysAgo'),'aphacker',2);
+        cy.visit(['sample', 'gradeable', team_gradeable, 'update?nav_tab=5']);
+        cy.get('#date_due')
+            .clear()
+            .type(getCurrentTime('twoDaysAgo'))
+            .type('{enter}');
+        cy.get('#save_status', {timeout:20000}).should('have.text', 'All Changes Saved');
+        cy.logout();
+        SubmitAndCheckMessage('team', 'upload_file1', 'valid_usage', '2_days_late+extention');
+    });
+    it('should show team warning message for the second submission ', () => {
+        /* Second submission happens after 3 days have passed. For student, it will be a valid submission,
+        so student will see a confirmation message first. However since aphacker doesn't have enough late days,
+        student will see a second warning message saying that aphacker will have a bad submission.*/
+        cy.login('instructor');
+        cy.visit(['sample', 'gradeable', team_gradeable, 'update?nav_tab=5']);
+        cy.get('#date_due')
+            .clear()
+            .type(getCurrentTime('threeDaysAgo'))
+            .type('{enter}');
+        cy.get('#save_status', {timeout:20000}).should('have.text', 'All Changes Saved');
+        cy.logout();
+        SubmitAndCheckMessage('team', 'upload_file2', 'both_messages', 'both_messages');
+    });
+
+    it('should cleanup everything that was added during testing', () => {
```

Comment:
Should this be in an 'after' block, instead of another test?

Reply:
When i put it in an 'after' block, it triggers ```cy.on('window:confirm', (t) => {}``` when it is not supposed to. So I will keep it in an 'it' block